tensorboard/components/tf_scalar_dashboard/tf-scalar-chart.html

```diff
@@ -0,0 +1,316 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../tf-backend/tf-backend.html">
+<link rel="import" href="../tf-color-scale/tf-color-scale.html">
+<link rel="import" href="../vz-line-chart/vz-line-chart.html">
+<link rel="import" href="../paper-icon-button/paper-icon-button.html">
+
+<!-- Prototype framework code. This will end up in its own file. -->
+<!--
+  A compact heading to appear above a single visualization, often
+  corresponding to either a single tag or a single run-tag combination.
+
+  Properties:
+    `title` does just what it says on the tin.
+    `color` can be set to display a colored border at the left of the
+        card; if left unset, no border will be displayed:
+    Any contents of the heading will be rendered below the title, and
+        can be used to display a subtitle, some small control widgets,
+        or similar.
+-->
+<dom-module id="tf-card-heading">
+  <template>
+    <div class="title-container" style="border-color: [[_borderColor]]">
+      <div
+        class="title"
+        inner-h-t-m-l="[[_break(title)]]"></div>
+      <div class="content">
+        <content></content>
+      </div>
+    </div>
+    <style>
+      .title-container {
+        border-left: 4px solid;
+        padding-left: 5px;
+        margin-bottom: 10px;
+      }
+      .title {
+        font-size: 14px;
+        text-overflow: ellipsis;
+        overflow: hidden;
+      }
+      .content {
+        font-size: 12px;
+      }
+    </style>
+  </template>
+  <script>
+    Polymer({
+      is: "tf-card-heading",
+      properties: {
+        title: String,
+        color: {
+          type: String,
+          value: null,   // this property is optional
+        },
+        _borderColor: {
+          type: String,
+          computed: '_computeBorderColor(color)',
+          readOnly: true,
+        },
+      },
+      _computeBorderColor(color) {
+        return color || 'rgba(255, 255, 255, 0.0)';  // 100% transparent white
+      },
+      _break: function(input) {
+        return input.replace(/([\/_-])/g, "$1<wbr>")
+      },
+    });
+  </script>
+</dom-module>
+
+<!--
+  A component that fetches scalar data from the TensorBoard server and
+  renders it into a vz-line-chart.
+-->
+<dom-module id="tf-scalar-chart">
+  <template>
+    <!--
+      Uncomment these to test the behavior of tf-card-heading.
+      They render like this:
+          card expanded:  http://i.imgur.com/Q7lbvxk.png
```

Comment:
Hmm…not for me?

![M E T A](https://user-images.githubusercontent.com/4317806/27247460-8230ac76-52ad-11e7-82f0-8771bb6fabd9.png)